homeassistant/components/kef/media_player.py

```diff
@@ -121,9 +108,7 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
         sources,
     )
 
-    mode = get_ip_mode(host)
-    mac = await hass.async_add_executor_job(partial(get_mac_address, **{mode: host}))
-    unique_id = f"kef-{mac}" if mac is not None else None
+    unique_id = f"kef-{speaker_type}-{host}-{port}"
```

Comment:
We don't allow ip address or hostname in the unique id.

https://developers.home-assistant.io/docs/entity_registry_index#unacceptable-sources-for-a-unique-id